Petroleum Plastics System

voidsong-dragonfly opened the pull request:
Added in components of petroleum plastics: Naphtha, Cracked Naphtha, Benzene, Propylene, Ethylene  
Added uses to components of petroleum plastics: portable generator, chemthrower, etc  
Added recipes for petroleum plastic feedstock to turn into duroplast  
Updated manual to reflect new recipe changes  
Reorganized manual to reflect the addition of several pages of content & to make it more intelligible  

Added 3 new advancements for petroleum plastic production
Added SRU manual pages & changed name to High-Pressure Refinery Unit

**NOT YET IMPLEMENTED**:
~~Advancements for any of the new content~~ Implemented as of 4th commit
~~Anything surrounding the SRU minus recipes - no manual updates, no recipe updates~~ Implemented as of 3rd commit
Anything surrounding the possible inclusion of hydrogen & proper hydrotreatment  
Anything surrounding rebalances of the portable generator's power output  

**POSSIBLE REQUIRED FIXES:**
Codestyle issues because I'm sure I've missed something  
Balance referendums from Twisted & possibly others

**GENERAL PROCESS DIAGRAM BELOW:**
![image](https://user-images.githubusercontent.com/40431605/185626052-8c6c15a9-b605-47d8-bb76-2953087c0cbb.png)

voidsong-dragonfly committed:
First pass of new petroleum plastics system

Added in components of petroleum plastics: Naphtha, Cracked Naphtha, Benzene, Propylene, Ethylene
Added uses to components of petroleum plastics: portable generator, chemthrower, etc
Added recipes for petroleum plastic feedstock to turn into duroplast
Updated manual to reflect new recipe changes
Reorganized manual to reflect the addition of several pages of content & to make it more intelligible

voidsong-dragonfly committed:
Fix up minor codestyle & typing error

voidsong-dragonfly committed:
Update SRU Manual & Change Name

- SRU -> HPRU, High Pressure Refinery Unit
- SRU manual entry is updated to reflect this
    - Others that mention the SRU have been updated
- Changed cracking recipe in SRU to be MV

voidsong-dragonfly committed:
Add advancements for PR

 - Added advancement for naphtha cracking
 - Added advancement for distilling the intermediates
 - Added advancement for getting duroplast from oil

voidsong-dragonfly committed:
Fix codestyle issues

TwistedGate commented on src/datagen/java/flaxbeard/immersivepetroleum/common/data/IPAdvancements.java:
```diff
@@ -5,12 +5,14 @@
-import net.minecraft.advancements.Advancement;
-import net.minecraft.advancements.AdvancementRewards;
-import net.minecraft.advancements.FrameType;
+import net.minecraft.advancements.*;
```
Disgusting! :D

voidsong-dragonfly committed:
Fix up * import

TwistedGate commented on src/main/resources/assets/immersivepetroleum/manual/en_us/hydrotreater.txt:
```diff
@@ -1,13 +1,19 @@
-Sulfur Recovery Unit
+High-Pressure Refinery Unit
 Smell of Fireworks
```
Since the name changed, the subtitle could need a snazzy change aswell.

voidsong-dragonfly replied:
Oh, good point!

voidsong-dragonfly committed:
Add preliminary compat structure

TwistedGate commented on src/main/resources/assets/immersivepetroleum/manual/en_us/napalm.txt:
```diff
@@ -1,3 +0,0 @@
-Napalm
-Beware the River God
-Napalm is an extremely flammable substance that can be created in the <link;mixer;§n§oMixer§r> using <link;immersivepetroleum:distillationtower;§n§oGasoline§r> and three measures of Aluminium Grit. It has little use other than wanton destruction; when Napalm is lit, fire will spread almost immediately over its surface, allowing for rapid deforestation. In addition, it can serve as a very effectiv ...
```
what, why is i gone, what happend? lol

voidsong-dragonfly replied:
It got merged into a bigger entry! See "Petroleum Products", labeled `fluids` in the text? It's in there now!

voidsong-dragonfly committed:
Fix up config name of HPRU

voidsong-dragonfly committed:
Finish fixing manual text for clarity

 - Added better links to the distillation column
 - Linked to reservoirs under petroleum
 - Linked to more uses within fluid entries
 - Added asphalt speed boost text to manual
 - Change HPRU secondary title text

TwistedGate commented on src/main/resources/assets/immersivepetroleum/manual/en_us/hydrotreater.txt:
```diff
@@ -1,5 +1,5 @@
 High-Pressure Refinery Unit
-Smell of Fireworks
+What's A Palladium Catalyst?
```
Who knows

TwistedGate reviewed (commented):
Looks fine to me

voidsong-dragonfly committed:
Add PneumaticCraft: Repressurized Compat

 - Add PNC:R to the build.gradle file from CurseMaven
 - Add recipes for PNC:R Plastic from Naphtha derivates
     - Propylene x100 -> Plastic x2000
     - Ethylene  x100 -> Plastic x1000
     - Small chance of bitumen "failed plastic" on each
